core/src/main/java/io/micronaut/core/util/localeresolution/LocaleResolutionConfiguration.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.util.localeresolution;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ * Locale resolution configuration.
+ *
+ * @author Sergio del Amo
+ * @since 2.3.0
+ */
+public interface LocaleResolutionConfiguration {
```

Comment:
I don't think we need a separate interface for this

Reply:
I think it is best to abstract with an interface. Then in an application, which does not use the http-server module (E.g. Alexa), I could implement the interface and support fixed locale and default resolution without effort.

Reply:
You an also do so without effort without this interface. The fixed and default can be direct args instead of having a separate interface

Reply:
If we have an interface, then it forces us to use the use keys in configuration. 

I could have: 

```
alexa:
  skills:
    myskill:
      skill-id 'xxxx-yaaa-zz123' 
      locale-resolution: 
          default-locale: 'es_ES'
```      

Having the interface forces us to reuse the same `default-locale` key in both `http-server` configuration and other configurations.

Reply:
If you want to leave the interface for use in other places thats OK, but the fixed resolver shouldn't require it. Users should be able to pull the data from anywhere and not have to conform to our contract

Reply:
I've splitted the configuration in two interfaces. Now, I am injecting only the `fixed` configuration in the `fixed` resolver.